satpy/composites/__init__.py

```diff
@@ -155,7 +156,11 @@ def match_data_arrays(self, data_arrays):
         """Match data arrays so that they can be used together in a composite."""
         self.check_geolocation(data_arrays)
         new_arrays = self.drop_coordinates(data_arrays)
-        if hasattr(xr, 'unify_chunks'):
+        # Only unify chunks for pure y/x DataArrays
+        # Other dimensions like 'bands' may have different sizes which is
+        # allowed and expected in certain cases. It is not possible to
+        # unify chunks in those cases.
+        if hasattr(xr, 'unify_chunks') and not any(set(x.dims) - {'y', 'x'} for x in data_arrays):
```

Comment:
This test could be its own method, with the comment above as docstring. That way we can give a readable name to the test.

Reply:
Good point. I might actually make this a utility function in the utils module since it may need to get reused. I also just realized I could make this less strict and available to more use cases (it is OK to have `bands` but only if they are all the same size).

Reply:
Done. I called it `satpy.utils.unify_chunks` instead of something longer since it is now a more generic utility function. Let me know what you think.